encoding/iid/VertexIID.java

```diff
@@ -233,44 +231,30 @@ private Attribute(ByteArray bytes) {
         }
 
         public static VertexIID.Attribute<?> of(ByteArray bytes) {
-            switch (Encoding.ValueType.of(bytes.get(PREFIX_W_TYPE_LENGTH))) {
-                case BOOLEAN:
-                    return new Attribute.Boolean(bytes);
-                case LONG:
-                    return new Attribute.Long(bytes);
-                case DOUBLE:
-                    return new Attribute.Double(bytes);
-                case STRING:
-                    return new Attribute.String(bytes);
-                case DATETIME:
-                    return new Attribute.DateTime(bytes);
-                default:
-                    assert false;
-                    throw TypeDBException.of(UNRECOGNISED_VALUE);
-            }
+            Encoding.ValueType<?> valueType = Encoding.ValueType.of(bytes.get(PREFIX_W_TYPE_LENGTH));
```

Comment:
replace `switch` on an `enum` with branches over instances of the encoding value type